test/unit_test/tutorial_test.cc

```diff
@@ -0,0 +1,196 @@
+#ifndef STATIC_LINK_VW
+#  define BOOST_TEST_DYN_LINK
+#endif
+
+#include <boost/test/unit_test.hpp>
+#include <boost/test/test_tools.hpp>
+
+#include "vw.h"
+#include "test_common.h"
+#include <numeric>
+#include <fmt/format.h>
+
+class Simulator
```

Comment:
should this class live on its own file and namespace? this way we can re-use in other unit tests.